cloud-event-management/lib/cloud-event-management-stack.ts

```diff
@@ -0,0 +1,132 @@
+import * as cdk from '@aws-cdk/core';
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as ecs from '@aws-cdk/aws-ecs';
+import * as rds from '@aws-cdk/aws-rds';
+import { AutoScalingGroup } from '@aws-cdk/aws-autoscaling';
+import { ApplicationLoadBalancer } from '@aws-cdk/aws-elasticloadbalancingv2';
+import * as acm from '@aws-cdk/aws-certificatemanager';
+
+export class CloudEventManagementStack extends cdk.Stack {
+  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    cdk.Tags.of(this).add('project', 'event-management');
+
+    const vpc = new ec2.Vpc(this, 'VPC', { natGateways: 0 });
+
+    //load balancer security group
+    const lbSG = new ec2.SecurityGroup(this, 'LB-SG', {
+      vpc,
+      description: 'Security group for the load balancer',
+    });
+
+    lbSG.addIngressRule(
+      ec2.Peer.anyIpv4(),
+      ec2.Port.tcp(443),
+      'Allow HTTP traffic for the load balancer'
+    );
+
+    const appSG = new ec2.SecurityGroup(this, 'App-SG', {
+      vpc,
+      description: 'Security group for the application',
+    });
+
+    appSG.addIngressRule(
+      lbSG,
+      ec2.Port.allTcp(),
+      'Allow traffic from the load balancer'
+    );
+
+    //db sg
+    const dbSG = new ec2.SecurityGroup(this, 'DB-SG', {
+      vpc,
+      description: 'SG for the application',
+    });
+
+    dbSG.addIngressRule(
+      appSG,
+      ec2.Port.tcp(3306),
+      'Allow My sql traffic from the application'
+    );
+
+    const asg = new AutoScalingGroup(this, 'ASG', {
+      vpc,
+      instanceType: ec2.InstanceType.of(
+        ec2.InstanceClass.T3,
+        ec2.InstanceSize.MICRO
+      ),
+      machineImage: ecs.EcsOptimizedImage.amazonLinux2(),
+      minCapacity: 1,
+      maxCapacity: 1,
+      securityGroup: appSG,
+      vpcSubnets: { subnets: vpc.publicSubnets },
+    });
+
+    //dbSecret.grantRead(asg.role);
+
+    const db = rds.DatabaseInstance.fromDatabaseInstanceAttributes(this, 'db', {
+      securityGroups: [dbSG],
+      port: 3306,
+      instanceEndpointAddress:
+        'event-management-db.c1i49kbi7ss8.eu-west-1.rds.amazonaws.com',
+      instanceIdentifier: 'event-management-db',
+    });
+
+    const cluster = new ecs.Cluster(this, 'EcsCluster', { vpc });
+    cluster.addAutoScalingGroup(asg);
+
+    // Create Task Definition
+    const taskDefinition = new ecs.Ec2TaskDefinition(this, 'TaskDef');
```

Comment:
The currently running task definition is called "event-management-task2" you can find it on ECS.